tests/utils/utils.go

```diff
@@ -323,6 +322,79 @@ func CreateReceiveCrossChainMessageTransaction(
 	return SignTransaction(destinationTx, senderKey, subnetInfo.EVMChainID)
 }
 
+// Constructs a transaction to call addProtocolVersion
+// Returns the signed transaction.
+func CreateAddProtocolVersionTransaction(
+	ctx context.Context,
+	signedMessage *avalancheWarp.Message,
+	teleporterRegistryAddress common.Address,
+	senderKey *ecdsa.PrivateKey,
+	subnetInfo interfaces.SubnetTestInfo,
+) *types.Transaction {
+	// Construct the transaction to send the Warp message to the destination chain
+	log.Info("Constructing addProtocolVersion transaction for the destination chain")
+	// numSigners, err := signedMessage.Signature.NumSigners()
+	// Expect(err).Should(BeNil())
+
+	callData, err := teleporterregistry.PackAddProtocolVersion(0)
+	Expect(err).Should(BeNil())
+
+	gasFeeCap, gasTipCap, nonce := CalculateTxParams(ctx, subnetInfo, PrivateKeyToAddress(senderKey))
+
+	destinationTx := predicateutils.NewPredicateTx(
+		subnetInfo.EVMChainID,
+		nonce,
+		&teleporterRegistryAddress,
+		500_000,
+		gasFeeCap,
+		gasTipCap,
+		big.NewInt(0),
+		callData,
+		types.AccessList{},
+		warp.ContractAddress,
+		signedMessage.Bytes(),
+	)
+
+	return SignTransaction(destinationTx, senderKey, subnetInfo.EVMChainID)
```

Comment:
I feel like this function could also call SendTransactionAndWaitForSuccess and return the receipt.

Reply:
going to leave this, similar to `CreateReceiveCrossChainMessageTransaction`, `CreateSendCrossChainMessageTransaction`, etc.